src/evict/evict_lru.c

```diff
@@ -2368,6 +2426,7 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
                 WT_RET_NOTFOUND_OK(__wt_tree_walk_count(session, &ref, &refs_walked, walk_flags));
         }
         btree->evict_ref = ref;
+        __evict_clear_walk(session, false);
     }
 
```

Comment:
Can we have an assertion here that looks something like - 
WT_ASSERT_ALWAYS(session, session->hazards.num_active == 0). 

It would be really if we exit this function with something pinned.

Reply:
@ershov can you respond to this?

Reply:
The `__evict_clear_walk` function clears the hazard pointer. Is there any reason why I need to check if it did it straight after calling it?